app.js

```diff
@@ -250,7 +250,12 @@ function gotMediaStream(stream) {
   var video = document.querySelector("video");
   video.src = URL.createObjectURL(stream);
   localStream = stream;
-  stream.onended = function() { console.log("Ended"); };
+  stream.getTracks().forEach(function(track) {
+    track.addEventListener('ended', function() {
+      console.log(stream.id, 'track ended', track.kind, track.id);
+      stopRecording();
```

Comment:
this line is actually the important thing. Everything else is cleaning up the lack of stream.onended.